tests/ipcMain/forkWorkspace.test.ts

```diff
@@ -0,0 +1,399 @@
+        const workspacePaths = projectConfig!.workspaces.map((ws) => ws.path);
+        expect(workspacePaths).toContain(sourceWorkspacePath);
+        expect(workspacePaths).toContain(forkedWorkspacePath);
+
+        // Cleanup
+        await env.mockIpcRenderer.invoke(IPC_CHANNELS.WORKSPACE_REMOVE, sourceWorkspaceId);
+        await env.mockIpcRenderer.invoke(IPC_CHANNELS.WORKSPACE_REMOVE, forkResult.metadata.id);
+      } finally {
+        await cleanupTestEnvironment(env);
+        await cleanupTempGitRepo(tempGitRepo);
+      }
+    },
+    30000
+  );
+
+  test.concurrent(
+    "should create independent git branch for forked workspace",
+    async () => {
+      const env = await createTestEnvironment();
+      const tempGitRepo = await createTempGitRepo();
+
+      try {
+        // Create source workspace
+        const trunkBranch = await detectDefaultTrunkBranch(tempGitRepo);
+        const createResult = await env.mockIpcRenderer.invoke(
+          IPC_CHANNELS.WORKSPACE_CREATE,
+          tempGitRepo,
+          "source-branch",
+          trunkBranch
+        );
+        expect(createResult.success).toBe(true);
+        const sourceWorkspaceId = createResult.metadata.id;
+
+        // Fork the workspace
+        const forkResult = await env.mockIpcRenderer.invoke(
+          IPC_CHANNELS.WORKSPACE_FORK,
+          sourceWorkspaceId,
+          "forked-branch"
+        );
+
+        expect(forkResult.success).toBe(true);
+
+        // Verify both workspaces exist and have different paths (use names for directory lookup)
+        const sourceInfo = await env.mockIpcRenderer.invoke(
+          IPC_CHANNELS.WORKSPACE_GET_INFO,
+          sourceWorkspaceId
+        );
+        expect(sourceInfo).toBeTruthy();
+
+        // Use fork result metadata directly (just returned, guaranteed to be current)
+        const sourceWorkspacePath = env.config.getWorkspacePath(tempGitRepo, sourceInfo.name);
+        const forkedWorkspacePath = env.config.getWorkspacePath(
+          tempGitRepo,
+          forkResult.metadata.name
+        );
+        expect(forkedWorkspacePath).not.toBe(sourceWorkspacePath);
+
+        // Verify both workspace directories exist
+        const sourceExists = await fs
+          .access(sourceWorkspacePath)
+          .then(() => true)
+          .catch(() => false);
+        const forkedExists = await fs
+          .access(forkedWorkspacePath)
+          .then(() => true)
+          .catch(() => false);
+
+        expect(sourceExists).toBe(true);
+        expect(forkedExists).toBe(true);
+
+        // Cleanup
+        await env.mockIpcRenderer.invoke(IPC_CHANNELS.WORKSPACE_REMOVE, sourceWorkspaceId);
+        await env.mockIpcRenderer.invoke(IPC_CHANNELS.WORKSPACE_REMOVE, forkResult.metadata.id);
+      } finally {
+        await cleanupTestEnvironment(env);
+        await cleanupTempGitRepo(tempGitRepo);
+      }
+    },
+    30000
+  );
+
+  test.concurrent(
+    "should handle existing partial.json when forking",
```

Comment:
These tests are a bit too coupled with implementation. Refer to your instructions on trying to test things as a user experiences them.